src/components/Dates/components/ExpYearInput.tsx

```diff
@@ -1,15 +1,14 @@
-import { ChangeEvent, Dispatch } from "react"
-import { FormActions } from "../../../reducers/form-reducer"
-import { formatYear } from "../../../helpers/helpers"
+import { ChangeEvent } from "react"
 import { EFormaActions, EFormData } from "../../../enums/Form"
+import { formatYear } from "../../../helpers/helpers"
+import { useForm } from "../../../hooks/useForm"
 
-type ExpYearInputProps = {
-    state: string;
-    dispatch: Dispatch<FormActions>;
-    error?: string;
-}
+const ExpYearInput = () => {
+
+    const { state, dispatch, errors } = useForm()
+    const { expYear } = state
+    const expYearError = errors[EFormData.ExpYear];
 
-const ExpYearInput = ({ state, dispatch, error }: ExpYearInputProps) => {
     const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
this handleChange should be ideal to have it in the context so you remove most of the logic to the context and keep clean the component